### PR TITLE
Pc/weekly calandar view

### PR DESCRIPTION
**Pull request by PeterChu2**
3 things:
1) replace list of components with a weekly calendar view in the detail view
2) when you click on the course item in the weekly calendar view, you can see all the text details
3) refactor ComponentView to take a CourseComponent rather than a long list of parameters

will upload screenshot of new UI.

### CODE REVIEW

**Comment by PeterChu2** [comment]
screen1

on click:
screen2

**Comment by PeterChu2** [comment]
can clean up UI later -- will work on adding this to the home screen to display all added courses

File: app/src/main/java/com/example/peterchu/watplanner/Models/Schedule/CourseComponent.java [hunk 1]
```diff
@@ -195,4 +207,79 @@ public void setInstructors(String[] instructors) {
...
+            Date eventEndTime = componentDateFormat.parse(this.endTime);
+            String eventName = String.format("%s %s %s", subject, catalogNumber, type);
+
+            Calendar date = Calendar.getInstance();
+            date.set(Calendar.MONTH, month);
+            date.set(Calendar.YEAR, 2017);
+            date.set(Calendar.DAY_OF_MONTH, 1);
+
+            Integer dayOfWeek = this.getDayOfWeek();
+            if (dayOfWeek == null || (month < Calendar.MAY) || (month > Calendar.AUGUST)) {
```
Comment (PeterChu2) [inline app/src/main/java/com/example/peterchu/watplanner/Models/Schedule/CourseComponent.java]: hack for current term LOL - should update with term data at some point

**Review by yongla — APPROVED** [review]

File: app/src/main/java/com/example/peterchu/watplanner/Models/Schedule/CourseComponent.java (hunk 1 above)
Comment (yongla) [reply app/src/main/java/com/example/peterchu/watplanner/Models/Schedule/CourseComponent.java]: Can you leave a comment about this? I'm a bit confused, but other than that this is good

**Pull request merged**